ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/parameter/HeaderParameterMapper.java

```diff
@@ -0,0 +1,198 @@
+            annotations = defaultableParameterNode.annotations();
+        }
+        if (Objects.isNull(annotations) || annotations.isEmpty()) {
+            return defaultName;
+        }
+        for (AnnotationNode annotation : annotations) {
+            if (annotation.annotReference().toString().trim().equals("http:Header")) {
+                String valueExpression = getNameFromHeaderAnnotation(annotation);
+                if (valueExpression != null) {
+                    return valueExpression;
+                }
+            }
+        }
+        return defaultName;
+    }
+
+    private static String getNameFromHeaderAnnotation(AnnotationNode annotation) {
+        Optional<MappingConstructorExpressionNode> annotationRecord = annotation.annotValue();
+        if (annotationRecord.isPresent()) {
+            SeparatedNodeList<MappingFieldNode> fields = annotationRecord.get().fields();
+            for (MappingFieldNode field : fields) {
+                if (field instanceof SpecificFieldNode specificFieldNode &&
+                        specificFieldNode.fieldName().toString().trim().equals("name")) {
+                        Optional<ExpressionNode> expressionNode = specificFieldNode.valueExpr();
+                        if (expressionNode.isPresent()) {
+                            ExpressionNode valueExpression = expressionNode.get();
+                            if (valueExpression.kind().equals(STRING_LITERAL)) {
+                                return valueExpression.toString().trim().replaceAll("\"", "");
+                            }
+                        }
+                }
+            }
+        }
+        return null;
+    }
+
+    public List<Parameter> getParameterSchemaList() {
+        if (Objects.isNull(type)) {
+            return new ArrayList<>();
+        }
+
+        List<Parameter> headerParameters = getRecordParameterSchema();
+        if (Objects.nonNull(headerParameters)) {
+            return headerParameters;
+        }
+
+        return List.of(getParameterSchema());
+    }
+
+    @Override
+    public HeaderParameter getParameterSchema() {
+        HeaderParameter headerParameter = new HeaderParameter();
+        headerParameter.setName(name);
+        if (isRequired && (!treatNilableAsOptional || !UnionTypeMapper.hasNilableType(type))) {
+            headerParameter.setRequired(true);
+        }
+        Schema typeSchema = typeMapper.getTypeSchema(type);
+        if (Objects.nonNull(defaultValue)) {
+            TypeMapperInterface.setDefaultValue(typeSchema, defaultValue);
+        }
+        headerParameter.setSchema(typeSchema);
+        headerParameter.setDescription(description);
+        return headerParameter;
+    }
+
+    public List<Parameter> getRecordParameterSchema() {
+        RecordTypeMapper.RecordTypeInfo recordTypeInfo = RecordTypeMapper.getDirectRecordType(type, null);
+        if (Objects.isNull(recordTypeInfo)) {
+            return null;
```

Comment:
Ideally we should avoid returning `null` from a method. In this case we could easily return and empty list.
```
```suggestion
            return List.of();
```